.github/workflows/build_and_deploy.yml

```
@@ -0,0 +1,131 @@
+name: Build Docker container and deploy app to AWS
+
+on:
+  push:
+    branches:
+      - develop
+      - main
+  workflow_dispatch:
+
+env:
+  ECR_REPOSITORY: alex-ic
+  REGION: ap-southeast-1
+
+jobs:
+  shared_variables:
+    name: Set env variables based on current branch
+    runs-on: ubuntu-latest
+
+    outputs:
+      env_name: ${{ steps.populate-outputs.outputs.env_name }}
+      container_name: ${{ steps.populate-outputs.outputs.container_name }}
+      ecs_cluster: ${{ steps.populate-outputs.outputs.ecs_cluster }}
+      ecs_service: ${{ steps.populate-outputs.outputs.ecs_service }}
+      ecs_task_definition: ${{ steps.populate-outputs.outputs.ecs_task_definition }}
+
+    steps:
+      - name: Set env BRANCH
+        run: echo "BRANCH=$(echo $GITHUB_REF | cut -d'/' -f 3)" >> $GITHUB_ENV
+
+      - name: Set current env variable
+        run: |
+          if [[ $BRANCH == 'main' ]]; then
+              echo "CURRENT_ENV=production" >> "$GITHUB_ENV"
+          else
+              echo "CURRENT_ENV=staging" >> "$GITHUB_ENV"
+          fi
+
+      - name: Populate outputs
+        id: populate-outputs
+        run: |
+            echo "env_name=${{ env.CURRENT_ENV }}" >> $GITHUB_OUTPUT
+            echo "container_name=alex-ic-${{ env.CURRENT_ENV }}" >> $GITHUB_OUTPUT
+            echo "ecs_cluster=alex-ic-${{ env.CURRENT_ENV }}-ecs-cluster" >> $GITHUB_OUTPUT
+            echo "ecs_service=alex-ic-${{ env.CURRENT_ENV }}-ecs-service" >> $GITHUB_OUTPUT
+            echo "ecs_task_definition=alex-ic-${{ env.CURRENT_ENV }}-service" >> "$GITHUB_OUTPUT"
+
+  build_image:
+    needs: shared_variables
+    name: Build the Docker image
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    outputs:
+      image: ${{ steps.build-image.outputs.image }}
+
+    steps:
+      - name: Debug
+        run: |
+          echo "ECR_REPOSITORY=${{ env.ECR_REPOSITORY }}"
+          echo "REGION=${{ env.REGION }}"
+          echo "env_name=${{ needs.shared_variables.outputs.env_name }}"
+          echo "container_name=${{ needs.shared_variables.outputs.container_name }}"
+          echo "ecs_cluster=${{ needs.shared_variables.outputs.ecs_cluster }}"
+          echo "ecs_service=${{ needs.shared_variables.outputs.ecs_service }}"
+          echo "ecs_task_definition=${{ needs.shared_variables.outputs.ecs_task_definition }}"
+      
+      - name: ENV variables
```

Follow Nimble's naming [convention](https://nimblehq.co/compass/development/code-conventions/github-actions/#convention-5), the name should start with a verb to describe the intention
```suggestion
      - name: Expose useful environment variables
```

---

Thanks. It was updated there: https://github.com/Nihisil/nimble-devops-ic-app/pull/12/commits/99cdd3bc7ba173579f52d4e202f69e41881de318